core/shared/src/main/scala/zio/ZSchedule.scala

```diff
@@ -713,14 +781,25 @@ object ZSchedule {
   final def logInput[R, A](f: A => ZIO[R, Nothing, Unit]): ZSchedule[R, A, A] =
     identity[A].logInput(f)
 
+  /**
+   * A schedule that waits forever when updating.
+   */
+  final val never: Schedule[Any, Unit] =
+    ZSchedule[Any, Nothing, Any, Unit](UIO.never, (_, _) => UIO.never, (_, _) => ())
```

Comment:
Return the `Nothing` that you get from the state, then you can make this `never` succeed with `Nothing` instead of `Unit`